File: app/src/main/java/de/karzek/diettracker/presentation/main/cookbook/recipeManipulation/adapter/viewHolder/RecipeManipulationManualIngredientItemViewHolder.java
package de.karzek.diettracker.presentation.main.cookbook.recipeManipulation.adapter.viewHolder;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.ViewGroup;
import android.widget.TextView;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import de.karzek.diettracker.R;
import de.karzek.diettracker.presentation.main.cookbook.recipeManipulation.adapter.itemWrapper.RecipeManipulationViewItemWrapper;
import de.karzek.diettracker.presentation.model.ManualIngredientDisplayModel;
import de.karzek.diettracker.presentation.util.StringUtils;

/**
 * Created by MarjanaKarzek on 30.05.2018.
 *
 * @author Marjana Karzek
 * @version 1.0
 * @date 30.05.2018
 */
public class RecipeManipulationManualIngredientItemViewHolder extends RecyclerView.ViewHolder {

    @BindView(R.id.ingredient_summary) TextView summary;

    private final OnDeleteIngredientClickListener onDeleteIngredientClickListener;
    private final OnManualIngredientClickedListener onManualIngredientClickedListener;

    public RecipeManipulationManualIngredientItemViewHolder(ViewGroup viewGroup,
                                                            OnDeleteIngredientClickListener onDeleteIngredientClickListener,
                                                            OnManualIngredientClickedListener onManualIngredientClickedListener) {
        super(LayoutInflater.from(viewGroup.getContext())
                .inflate(R.layout.viewholder_recipe_ingredient, viewGroup, false));
        ButterKnife.bind(this, itemView);

        this.onDeleteIngredientClickListener = onDeleteIngredientClickListener;
        this.onManualIngredientClickedListener = onManualIngredientClickedListener;
    }

    public void bind(RecipeManipulationViewItemWrapper item, int id) {
        summary.setText(formatSummary((ManualIngredientDisplayModel) item.getIngredientDisplayModel()));
        itemView.setTag(id);
    }

    private String formatSummary(ManualIngredientDisplayModel ingredient){
        String amount = StringUtils.formatFloat(ingredient.getAmount());
        return amount + ingredient.getUnit().getName() + " " + ingredient.getGroceryQuery();
    }

    @OnClick(R.id.action_delete) public void onDeleteIngredientClicked() {
        onDeleteIngredientClickListener.onDeleteManualIngredientClicked((int) itemView.getTag());
    }

    @OnClick(R.id.item_layout) public void setOnManualIngredientClicked() {
        onManualIngredientClickedListener.onManualIngredientClicked((int) itemView.getTag());
    }

    public interface OnDeleteIngredientClickListener {
        void onDeleteManualIngredientClicked(int id);
    }

    public interface OnManualIngredientClickedListener {
        void onManualIngredientClicked(int id);
    }

}

File: app/src/main/java/de/karzek/diettracker/presentation/main/cookbook/recipeManipulation/adapter/viewHolder/RecipeManipulationManualIngredientItemViewHolder.kt
package de.karzek.diettracker.presentation.main.cookbook.recipeManipulation.adapter.viewHolder

import android.view.View
import androidx.recyclerview.widget.RecyclerView
import de.karzek.diettracker.presentation.main.cookbook.recipeManipulation.adapter.itemWrapper.RecipeManipulationViewItemWrapper
import de.karzek.diettracker.presentation.model.ManualIngredientDisplayModel
import de.karzek.diettracker.presentation.util.StringUtils
import kotlinx.android.extensions.LayoutContainer
import kotlinx.android.synthetic.main.viewholder_recipe_ingredient.*

/**
 * Created by MarjanaKarzek on 30.05.2018.
 *
 * @author Marjana Karzek
 * @version 1.0
 * @date 30.05.2018
 */
class RecipeManipulationManualIngredientItemViewHolder(override val containerView: View,
                                                       private val onDeleteIngredientClickListener: OnDeleteIngredientClickListener,
                                                       private val onManualIngredientClickedListener: OnManualIngredientClickedListener) : RecyclerView.ViewHolder(containerView), LayoutContainer {

    fun bind(item: RecipeManipulationViewItemWrapper, id: Int) {
        ingredient_summary.text = formatSummary(item.ingredientDisplayModel as ManualIngredientDisplayModel)
        itemView.tag = id
        setupClickListeners()
    }

    private fun setupClickListeners() {
        action_delete.setOnClickListener { onDeleteIngredientClickListener.onDeleteManualIngredientClicked(itemView.tag as Int) }
        item_layout.setOnClickListener { onManualIngredientClickedListener.onManualIngredientClicked(itemView.tag as Int) }
    }

    private fun formatSummary(ingredient: ManualIngredientDisplayModel): String {
        val amount = StringUtils.formatFloat(ingredient.amount)
        return amount + ingredient.unit.name + " " + ingredient.groceryQuery
    }

    interface OnDeleteIngredientClickListener {
        fun onDeleteManualIngredientClicked(id: Int)
    }

    interface OnManualIngredientClickedListener {
        fun onManualIngredientClicked(id: Int)
    }

}
